core/src/main/scala/cromwell/core/io/IoCommand.scala

```diff
@@ -38,6 +38,12 @@ trait IoCommand[+T] {
 
   def commandDescription: String
 
+  /**
+    * IO commands side-effect and/or have exceptions. We don't want that when evaluating identity, e.g. to check presence in cache.
+    * @return Hash code based on the description, which captures file path & action
+    */
+  override def hashCode(): Int = commandDescription.hashCode
```

Comment:
I remember you were thinking before about adding the UUID here as well. Any particular reason you decided against it? Do we *want* separate IOCommands that are doing exactly the same thing to have the same hash code?

Reply:
The UUID is randomly generated and I think it would prevent the cache from working in the case of otherwise-identical commands (same action, same file path). I'm not actually sure the cache is useful at all, but if we choose to stop using it, we should delete it.

One thing I tested in Ammonite is whether mixing in a trait with a UUID field feeds into the default Scala hashcode implementation and the answer is no, so we are not removing anything by no longer considering it.